compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirNotImplementedOverrideChecker.kt

```diff
@@ -0,0 +1,287 @@
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.fir.scopes.*
+import org.jetbrains.kotlin.fir.scopes.impl.unwrapDelegateTarget
+import org.jetbrains.kotlin.fir.symbols.CallableId
+import org.jetbrains.kotlin.fir.symbols.impl.*
+import org.jetbrains.kotlin.fir.types.*
+import org.jetbrains.kotlin.name.Name
+
+object FirNotImplementedOverrideChecker : FirClassChecker() {
+    // TODO: differentiate members with different annotations, e.g., @Api...(x) @Api...(y)
+    // TODO: differentiate parameter names? e.g., foo(x : T) v.s. override foo(y : T)
+    private object SignaturePresenter : FirDeclarationPresenter {
+        override fun StringBuilder.appendRepresentation(it: CallableId) {
+            append(it.callableName)
+        }
+    }
+
+    private fun FirClass<*>.unsubstitutedScope(context: CheckerContext) =
+        this.unsubstitutedScope(context.sessionHolder.session, context.sessionHolder.scopeSession, withForcedTypeCalculator = true)
+
+    // TODO: Collection extensions are not properly regarded as an override. Should not have a whitelist like this.
+    private val COLLECTIONS = listOf("java.util", "kotlin.collections")
+
+    override fun check(declaration: FirClass<*>, context: CheckerContext, reporter: DiagnosticReporter) {
+        // TODO: kt4763Property: reporting on `object` literal causes invalid error in test...FirDiagnosticHandler
+        if (declaration is FirAnonymousObject) return
+
+        val source = declaration.source ?: return
+        if (source.kind is FirFakeSourceElementKind) return
+
+        val contributedMembers = collectCallableMembers(declaration, context)
+        val potentialFakeOverrides = getBaseDeclarationsForFakeOverrides(declaration, contributedMembers, context)
+
+        // TODO: consider open as overridable, and check conflicts as well
+        val abstractFakeOverrides = potentialFakeOverrides.filter {
+            it is FirMemberDeclaration &&
+                    it.isAbstract &&
+                    // TODO: FIR MPP support
+                    (it.getContainingClass(context) as? FirRegularClass)?.isExpect == false
+        }
+        if (abstractFakeOverrides.isEmpty()) return
+
+        // TODO: differentiate type-substituted declarations. Otherwise, they will be reported as MANY_IMPL_MEMBER_NOT_IMPLEMENTED
+        val sigToDeclarations = mutableMapOf<String, MutableList<FirCallableDeclaration<*>>>()
+        for (abstractFakeOverride in abstractFakeOverrides) {
+            val sig = when (abstractFakeOverride) {
+                is FirSimpleFunction -> SignaturePresenter.represent(abstractFakeOverride)
+                is FirProperty -> SignaturePresenter.represent(abstractFakeOverride)
+                else -> continue
+            }
+            sigToDeclarations.computeIfAbsent(sig) { mutableListOf() }.add(abstractFakeOverride)
+        }
+
+        val canHaveAbstractMembers = declaration is FirRegularClass && declaration.canHaveAbstractFakeOverride
```

Comment:
If the class can have abstract members, shouldn't we just return? Nothing to do in that case.

Reply:
There are more things to check than just `abstract` members in this checker. E.g., if a class inherits from a multiple interfaces with default impl with the conflict signature, we need to catch that as well.

Reply:
Ah makes sense, thanks for clarifying! Somehow missed the whole TODO block. 🙂

Reply:
Yep, I disabled `MANY_*` diagnostics for now, and also noticed there could be many more relevant diagnostics.